src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -106,7 +107,28 @@ public User addUser(PrimaryKey id) {
 	
 	public ResponseEntity<?> getTopFavs(int page, int nums) {
 		List<Product> products = productRepository.findTopN(PageRequest.of(page, nums, Sort.by("numberOfFavorites").descending()));
-		return ResponseEntity.status(HttpStatus.OK).body(products);		
+		if(products == null) {
```

Comment:
Maybe let's do `(products ==null || products.isEmpty()) ` here

I think the default been returned when we have no records in our database for that page range is an empty list, but instead of returning from this IF clause, we still go down and try to call search APIs then finally return in line [122](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/pull/32/files#diff-0fae860ae0c1e35be827d26bc884c318R122)

I tested using a ridiculous page range as shown below:
https://hist-fav-checkout-beta.herokuapp.com/getTopFavs?page=1000000000&nums=100000

Please lmk if I'm wrong.

Reply:
Good catch. I totally agree. We should do an eager return. I will make the change and deploy. Thanks.